### PR TITLE
Feat(Species Trait Index): Adds modals when clicking on traits

### PR DESCRIPTION
**Pull request by AW0005**
Adds the ability to have modals pop up when you click on a trait on the trait index with the traits description instead of navigating to a separate page.

### CODE REVIEW

File: resources/views/world/species_features.blade.php [hunk 1]
```diff
@@ -11,7 +11,7 @@
     <p>This is a visual index of all {!! $species->displayName !!}-specific traits. Click a trait to view more info on it!</p>
 
     @foreach ($features as $categoryId => $categoryFeatures)
-        @if ($categories[$categoryId]->is_visible)
+        @if (!isset($categories[$categoryId]) || $categories[$categoryId]->is_visible)
```
Comment (SpeedyD) [inline resources/views/world/species_features.blade.php:14]: Apologies, but I don't understand this change? Could you explain this?

File: resources/views/world/species_features.blade.php [hunk 2]
```diff
@@ -41,3 +41,17 @@
         @endif
     @endforeach
 @endsection
+
+@section('scripts')
+@if(Config::get('lorekeeper.extensions.species_trait_index.trait_modals'))
```
Comment (SpeedyD) [inline resources/views/world/species_features.blade.php:46]: Very, very minor note, but could you perhaps tab the @if and inside the @if section? ^^;;

File: resources/views/world/species_features.blade.php (hunk 1 above)
Comment (AW0005) [reply resources/views/world/species_features.blade.php:14]: It's possible for traits to not have any category (and they're labeled with the "Miscellaneous" label on the lines just below this)

Comment (SpeedyD) [reply resources/views/world/species_features.blade.php:14]: ...so now I have two questions: How did I miss this when I added the is_visible check and how did it go through after a collective check XD

Thanks for fixing this!

**Review by SpeedyD — APPROVED** [review]

**Review by itinerare — APPROVED** [review]

**Pull request merged**